rider/views.py

```diff
@@ -15,148 +15,160 @@
+
+    # Check that the rider is in the group
+    rider_in_group_test = Group.objects.filter(rider__id=r_id).filter(code=aff_id)
+    if not rider_in_group_test:
+        return write_response(request, json.dumps([{"success": "false", "message": "ERROR: Not in group"}]))
+
+    # Get the numerical ID that matches the group's affinity code
+    # Then find the mapping entry in the Affinity Group Mapping table and delete it
+    group_id=Group.objects.get(code=aff_id).id
+    agm = Affinity_Group_Mapping.objects.filter(rider_id=r_id).filter(affinity_group_id=group_id)
+    agm.delete()
+
+    # return a success response - need to include the callback key for JSONP requests
+    return write_response(request, json.dumps([{"success": "true", "message": "Success"}]))
+
 
 def check_code_view(request, aff_id):	
-	#check if the code is in use
-	#if group_test doesn't come back empty return a 400 error code
-	group_test = Group.objects.filter(code=aff_id)
-	if group_test:
-		return write_response(request, json.dumps([{"success": "false", "message": "Code exists"}]))
-	return write_response(request, json.dumps([{"success": "true", "message": "Code does not exist"}]))
+    # check if the code is in use
+    # if group_test doesn't come back empty return a 400 error code
+    group_test = Group.objects.filter(code=aff_id)
+    if group_test:
+        return write_response(request, json.dumps([{"success": "false", "message": "Code exists"}]))
+    return write_response(request, json.dumps([{"success": "true", "message": "Code does not exist"}]))
+
 
 def write_response(request, data):
-	response = HttpResponse()
-	response.content_type="application/json"
+    response = HttpResponse()
+    response.content_type="application/json"
+
+    if 'callback' in request.REQUEST:
+        return_string = '%s(%s)' % (request.REQUEST['callback'], data)
+        response.write(return_string)
+        return response
+    response.write(data)
+    return response
 
-	if 'callback' in request.REQUEST:
-		return_string = '%s(%s)' % (request.REQUEST['callback'], data)
-		response.write(return_string)
-		return response
-	response.write(data)
-	return response
 
 class RiderAPI(APIView):
 
     def post(self, request, format=None):
         try:
-            logger.debug('trying')
-            data = request.DATA
-            serializer = riderSerializer(data=data)
-            if serializer.is_valid():
-                serializer.save()
-                rider_uuid = create_uuid(serializer.object.pk)
-
-                return Response(
-                    #{settings.JSON_KEYS['RIDER_ID']: rider_uuid},
-                    {settings.JSON_KEYS['RIDER_ID']: serializer.object.pk},
-		    status=status.HTTP_201_CREATED
-                )
+            data = request.data
+            logger.debug('[RiderAPI:Register] Attempting to find rider with id: '+str(data[u'id']))
+            rider_data = Rider.objects.filter(id=data[u'id']).exists()
```

Comment:
I'm not sure about the standard; but this, and the following if statement aren't as clear as they could be. This variable should be rider_data_exists so it would look like
```
if rider_data_exists:
```